### PR TITLE
Add difference imaging steps to LATISS DRP.yaml

### PR DESCRIPTION
**Pull request by hlin730**
Add difference imaging tasks in step4, step5, and step6.  Also modify step2bcde (move transformSourceTable, consolidateSourceTable, writeRecalibratedSourceTable from step4 to here) and step3b (add selectGoodSeeingVisits and templateGen).

### CODE REVIEW

**Review by ctslater — COMMENTED** [review]
I added a few thoughts to the yaml file; I think everything should work ok but there are a few tasks that might be better in different steps long term? I'm not super committed to that though, so open to discussion about these.

File: pipelines/LATISS/DRP.yaml [hunk 1]
```diff
@@ -178,6 +181,8 @@ subsets:
   step3b:
     subset:
       - consolidateObjectTable
+      - selectGoodSeeingVisits
+      - templateGen
```
[the remark this answers is not in the document]
Comment (hlin730) [reply pipelines/LATISS/DRP.yaml:185]: Agreed, I'll move these two tasks to step3a, same as for the HSC DRP-RC2.yaml and I believe for DP0.2.

File: pipelines/LATISS/DRP.yaml [hunk 2]
```diff
@@ -223,3 +232,43 @@ subsets:
...
+      - consolidateFullDiaObjectTable
+      - writeForcedSourceOnDiaObjectTable
+      - transformForcedSourceOnDiaObjectTable
+      - consolidateForcedSourceOnDiaObjectTable
+    description: |
+      Tasks that can be run together, but only after the 'step1', 'step2',
+      'step3', and 'step4' subsets.
+
+      This step includes patch-level aggregation Tasks. These should be run
+      with explicit 'tract' constraints in the data query, otherwise quanta
```
[the remark this answers is not in the document]
Comment (hlin730) [reply pipelines/LATISS/DRP.yaml:254]: Agreed about this not relevant for LATISS.  I'll remove this sentence.

File: pipelines/LATISS/DRP.yaml [hunk 3]
```diff
@@ -135,7 +135,10 @@ subsets:
       - fgcmBuildFromIsolatedStars
       - fgcmFitCycle
       - fgcmOutputProducts
+      - transformSourceTable
+      - consolidateSourceTable
       - updateVisitSummary
+      - writeRecalibratedSourceTable
```
[the remark this answers is not in the document]
Comment (hlin730) [reply pipelines/LATISS/DRP.yaml:141]: Looking more carefully at the HSC DRP-RC2.yaml, I see it has transformSourceTable in step4 and  consolidateSourceTable in step6.  For consistency, I can move these 2 tasks, plus writeRecalibratedSourceTable, to where they are in HSC DRP-RC2.yaml.

[the remark this answers is not in the document]
Comment (laurenam) [reply pipelines/LATISS/DRP.yaml]: As I've said on Slack, my motivation was based on having the SFM tables available as soon as possible for any QA scripts that use them as input.  I was also paying more attention to HSC's DRP-Prod.yaml pipeline (as I _think_ that is the one we should be taking more as standard).  There, the `fgcm` tasks are in `step2c` and `step2d` includes:
```  step2d:
    subset:
      - finalizeCharacterization
      - writeRecalibratedSourceTable
      - transformSourceTable
      - consolidateSourceTable
      - updateVisitSummary
```

[the remark this answers is not in the document]
Comment (hlin730) [reply pipelines/LATISS/DRP.yaml]: @laurenam Sorry I misunderstood things.  I didn't know about DRP-Prod.yaml.  If we want these 3 tasks moved to step2bcde, what do  you recommend we do at this point?

[the remark this answers is not in the document]
Comment (laurenam) [reply pipelines/LATISS/DRP.yaml]: I'm sure it's fine as is for now.  We haven't been running any visit-level `analysis_tools` regularly yet so, in that sense, I guess no one is in any particular hurry to have them available ASAP.  We can always make another ticket to move them if it becomes desirable (and I can ask around pipelines folk to see if they see any reason not to).

**Pull request merged**